python/cudf_polars/docs/overview.md

```diff
@@ -144,11 +144,69 @@ changes. We can therefore attempt to detect the IR version
 appropriately. This should be done during IR translation in
 `translate.py`.
 
-## Adding a handler for a new plan node
+# IR design
+
+As noted, we translate the polars DSL into our own IR. This is both so
+that we can smooth out minor version differences (advertised by
+`NodeTraverser` version changes) within `cudf-polars`, and so that we
+have the freedom to introduce new IR nodes and rewrite rules as might
+be appropriate for GPU execution.
+
+To that end, we provide facilities for definition of nodes as well as
+writing traversals and rewrite rules. The abstract base class `Node`
+in `dsl/nodebase.py` defines the interface for implementing new nodes,
+and provides many useful default methods. See also the docstrings of
+the `Node` class.
+
+> ![NOTE] This generic implementation relies on nodes being treated as
+> *immutable*. Do not implement in-place modification of nodes, bad
+> things will happen.
```

Comment:
What syntax is this? I haven't seen this kind of admonition before. rst uses the `.. ` syntax, whereas myst uses [this syntax](https://mystmd.org/guide/admonitions).

Reply:
Oh, it's github-flavoured markdown. Let me change it.